sdks/python/apache_beam/testing/benchmarks/nexmark/nexmark_launcher.py

```diff
@@ -0,0 +1,242 @@
+          --loglevel=DEBUG (optional) \
+          --sdk_location <apache_beam tar.gz> \
+          --staging_location=gs://... \
+          --temp_location=gs://
+
+"""
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import argparse
+import collections
+import logging
+import sys
+import uuid
+
+from google.cloud import pubsub
+
+import apache_beam as beam
+from apache_beam.options.pipeline_options import GoogleCloudOptions
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.options.pipeline_options import SetupOptions
+from apache_beam.options.pipeline_options import StandardOptions
+from apache_beam.options.pipeline_options import TestOptions
+from apache_beam.testing.benchmarks.nexmark.nexmark_util import Command
+from apache_beam.testing.benchmarks.nexmark.queries import query0
+
+
+class NexmarkLauncher(object):
+  def __init__(self):
+    self.parse_args()
+    self.uuid = str(uuid.uuid4())
+    self.topic_name = self.args.topic_name + self.uuid
+    self.subscription_name = self.args.subscription_name + self.uuid
+
+  def parse_args(self):
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--query', '-q',
+                        type=int,
+                        action='append',
+                        required=True,
+                        choices=[0, 1, 2],
+                        help='Query to run')
+
+    parser.add_argument('--subscription_name',
+                        type=str,
+                        help='Pub/Sub subscription to read from')
+
+    parser.add_argument('--topic_name',
+                        type=str,
+                        help='Pub/Sub topic to read from')
+
+    parser.add_argument('--loglevel',
+                        choices=['DEBUG', 'INFO', 'WARNING',
+                                 'ERROR', 'CRITICAL'],
+                        default='INFO',
+                        help='Set logging level to debug')
+    parser.add_argument('--input',
+                        type=str,
+                        required=True,
+                        help='Path to the data file containing nexmark events.')
+
+    self.args, self.pipeline_args = parser.parse_known_args()
+    logging.basicConfig(level=getattr(logging, self.args.loglevel, None),
+                        format='(%(threadName)-10s) %(message)s')
+
+    self.pipeline_options = PipelineOptions(self.pipeline_args)
+    logging.debug('args, pipeline_args: %s, %s', self.args, self.pipeline_args)
+
+    # Usage with Dataflow requires a project to be supplied.
+    self.project = self.pipeline_options.view_as(GoogleCloudOptions).project
+    if self.project is None:
+      parser.print_usage()
+      print(sys.argv[0] + ': error: argument --project is required')
+      sys.exit(1)
+
+    # Pub/Sub is currently available for use only in streaming pipelines.
+    self.streaming = self.pipeline_options.view_as(StandardOptions).streaming
```

Comment:
I was suggesting auto-adding the streaming flag. It is fine to not do this. In that, your usage example needs to be updated to add this flag.

Reply:
Done.